dnas/sensemaker/zomes/coordinator/sensemaker/src/assessment.rs

```diff
@@ -35,37 +50,74 @@ pub fn get_assessments_for_resource(
 pub struct CreateAssessmentInput {
     pub value: RangeValue,
     pub dimension_eh: EntryHash,
-    pub subject_eh: EntryHash, // assuming this is the EH of the resource being assessed
+    pub resource_eh: EntryHash,
+    pub resource_type_eh: EntryHash,
     pub maybe_input_dataset: Option<DataSet>,
 }
 
 #[hdk_extern]
-pub fn create_assessment(
-    CreateAssessmentInput {
-        value,
-        dimension_eh,
-        subject_eh,
-        maybe_input_dataset,
-    }: CreateAssessmentInput,
-) -> ExternResult<EntryHash> {
+pub fn create_assessment(CreateAssessmentInput { value, dimension_eh, resource_eh, resource_type_eh, maybe_input_dataset }: CreateAssessmentInput) -> ExternResult<EntryHash> {
     let assessment = Assessment {
         value,
         dimension_eh,
-        subject_eh,
+        resource_eh,
+        resource_type_eh,
         maybe_input_dataset,
         author: agent_info()?.agent_latest_pubkey,
     };
     create_entry(&EntryTypes::Assessment(assessment.clone()))?;
     let assessment_eh = hash_entry(&EntryTypes::Assessment(assessment.clone()))?;
+    let assessment_path = assessment_typed_path(assessment.resource_eh.clone(), assessment.dimension_eh)?;
+    // ensure the path components are created so we can fetch child paths later
+    assessment_path.clone().ensure()?;
     create_link(
-        assessment_typed_path(assessment.subject_eh, assessment.dimension_eh)?.path_entry_hash()?,
+        assessment_path.path_entry_hash()?,
         assessment_eh.clone(),
         LinkTypes::Assessment,
         (),
     )?;
     Ok(assessment_eh)
 }
 
+#[hdk_extern]
+pub fn get_all_assessments(_:()) -> ExternResult<Vec<AssessmentWithDimensionAndResource>> {
+    let base_path = all_assessments_typed_path()?;
+    let assessed_resources_typed_paths = base_path.children_paths()?;
+    let mut all_assessments: Vec<Vec<AssessmentWithDimensionAndResource>> = vec![];
+
+    // for each resource that has been assessed, crawl all children to get each dimension that it has been assessed along
+    for assessed_resource_path in assessed_resources_typed_paths {
+        let assessed_dimensions_for_resource_typed_paths = assessed_resource_path.children_paths()?;
+        
+        // for each dimension that a resource has been assessed, get the assessment
+        for assessed_dimension_path in assessed_dimensions_for_resource_typed_paths {
+            let assessments = get_links(assessed_dimension_path.path_entry_hash()?, LinkTypes::Assessment, None)?.into_iter().map(|link| {
```

Comment:
This `get_links` should be moved out of the for loop, no?

Reply:
@weswalla @julioholon I think what we can do here, (maybe in the future given this is not an urgent concern now), is to use holochain's parallel call (probably not the right name) in order to make the retrieval of these links faster. Something like https://github.com/holochain-open-dev/profiles/blob/main/crates/coordinator/src/lib.rs#L163-#L178 is what I am imagining we can do. This may not be the only improvement we can make but it certainly could make network calls like `get_links` return results faster than calling it sequentially.

Reply:
Yes, we can leave this for the future, as it's just a way to improve performance, and optimizing too early means releasing less often :)